tests/src/integration/array.rs

```diff
@@ -0,0 +1,4 @@
+#[test]
+fn binary_works() {
+    assert!(crate::integration::run_php("array.php"));
```

Comment:
> Not a 100% happy how the PHP code is executed using a hardcoded path to the compiled extension. Nontheless, I think that is a good start for the first iteration

And idea that might work is to put together a simple macro and embed the php snippet directly into this file:

```rust
#[test]
fn binary_works() {
    php_test!(r#"
// Tests sequential arrays
$array = test_array(['a', 'b', 'c']);

assert(is_array($array));
assert(count($array) === 3);
assert(in_array('a', $array));
assert(in_array('b', $array));
assert(in_array('c', $array));

// Tests associative arrays
$assoc = test_array_assoc([
    'a' => '1',
    'b' => '2',
    'c' => '3'
]);

assert(array_key_exists('a', $assoc));
assert(array_key_exists('b', $assoc));
assert(array_key_exists('c', $assoc));
assert(in_array('1', $assoc));
assert(in_array('2', $assoc));
assert(in_array('3', $assoc));
    "#);
}
```
For the initial version, we could either automatically write out a file an execute it or - I believe this should work - pipe the input into php without touching the filesystem.

I will resume working on my execution support, after which this could mature into something entirely in process.